File: bongf/java/src/level1/ReverseTernary.java
package level1;

public class ReverseTernary {
    public static void main(String[] args) {
        System.out.println(solution2(125));
    }

    public static int solution(int n) {
        String result = "";
        while(n > 0) {
            result += n % 3;
            n = n / 3;
        }

        int answer = 0;
        for(int i=0; i<result.length(); i++) {
            answer += (Integer.parseInt(String.valueOf(result.charAt(i)))) * Math.pow(3, result.length() - 1 - i);

        }
        return answer;
    }

    public static int solution2(int n) {
        String result = "";
        while(n > 0) {
            result += n % 3;
            n = n / 3;
        }
        return Integer.parseInt(result, 3);
    }
}

File: programmers/python/level1/reverse_ternary.py

def solution(n):
    result = ''
    while n > 0 :
        n, mod = divmod(n, 3)
        result += str(mod)
    
    while result[0] == '0' :
        result = result[1:]

    return sum( int(result[i]) * ( 3 ** (len(result) - 1 - i )) for i in range(len(result)))


def solution_2(n):
    result = ''
    while n:
        result += str(n%3)
        n = n//3 
    
    return int(result, 3)

print(solution_2(125))